agrona/src/main/java/org/agrona/concurrent/errors/DistinctErrorLog.java

```diff
@@ -157,38 +157,83 @@ public Charset charset()
      */
     public boolean record(final Throwable observation)
     {
-        final long timestampMs;
-        DistinctObservation distinctObservation;
+        final long timestampMs = clock.time();
```

Comment:
I think the above logic should be moved into the updateLastObservationTimestamp method.